background/main.js

```diff
@@ -1,11 +1,22 @@
 var SERVER_URL = "https://chronicle.dev.mozaws.net";
 var CREATE_VISIT_ENDPOINT = SERVER_URL + "/v1/visits";
 
-function createVisit(historyItem) {
-  console.log('create visit', historyItem);
+function createVisit(tabId, changeInfo, tab) {
+  // Filter out some visits:
```

Comment:
I'd like to separate the browser-specific logic from the shared logic. Suppose we had a browser abstraction that exposed, say, `browser.onTabLoaded` and `browser.request` methods, and a shared library with `createVisit` and `filterUrl` methods. That would let us share most of the logic across all the addons/extensions, and clearly delineate the boundaries between business logic and the underlying browser environment.

That said, I'm fine with just landing this code as-is for now, and filing a followup bug for cleanup and pretty abstractions :-)